File: java/src/test/java/org/sammancoaching/SecurityClearanceAuthorizerTest.java
package org.sammancoaching;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.sammancoaching.testing.HttpServerLifecycle;
import org.sammancoaching.testing.LightweightHttpServer;

import static org.junit.jupiter.api.Assertions.*;

@ExtendWith(HttpServerLifecycle.class)
public class SecurityClearanceAuthorizerTest {

    @Test
    void testAuthorize() {
        LightweightHttpServer webserver = HttpServerLifecycle.getHttpServer();
        SecurityClearanceAuthorizer authorizer = new SecurityClearanceAuthorizer(webserver.getUrl());
        webserver.getHandler().setResponse(200, "OK");

        boolean result = authorizer.isAuthorized();

        assertTrue(result);
        assertEquals("GET", webserver.getHandler().getLatestRequestType());
    }

    @Test
    void testNotAuthorize() {
        LightweightHttpServer webserver = HttpServerLifecycle.getHttpServer();
        SecurityClearanceAuthorizer authorizer = new SecurityClearanceAuthorizer(webserver.getUrl());
        webserver.getHandler().setResponse(404, "Not Found");

        boolean result = authorizer.isAuthorized();

        assertFalse(result);
        assertEquals("GET", webserver.getHandler().getLatestRequestType());
    }
}

File: kotlin/src/test/kotlin/org/sammancoaching/phonebook/SecurityClearanceAuthorizerTest.kt
package org.sammancoaching.phonebook

import org.junit.jupiter.api.extension.ExtendWith
import org.sammancoaching.testing.HttpServerLifecycle
import org.sammancoaching.testing.LightweightHttpServer
import kotlin.test.Test
import kotlin.test.assertEquals

@ExtendWith(HttpServerLifecycle::class)
class SecurityClearanceAuthorizerTest {
    @Test
    fun whenRemoteServiceReturns200_isAuthorized() {
        val server = HttpServerLifecycle.httpServer!!
        val authorizer = SecurityClearanceAuthorizer(server.url)

        val isAuthorized = authorizer.isAuthorized

        assertEquals(true, isAuthorized)
    }

    @Test
    fun whenRemoteServiceReturns404_isNotAuthorized() {
        val server = HttpServerLifecycle.httpServer!!
        server.getHandler().setResponse(404, "Not found")
        val authorizer = SecurityClearanceAuthorizer(server.url)

        val isAuthorized = authorizer.isAuthorized

        assertEquals(false, isAuthorized)
    }
}